modules/streams-core/src/main/scala/com.snowplowanalytics.snowplow/streams/Sink.scala

```diff
@@ -23,11 +23,21 @@ trait Sink[F[_]] {
   def sinkSimple(batch: ListOfList[Array[Byte]]): F[Unit] =
     sink(batch.mapUnordered(Sinkable(_, None, Map.empty)))
 
-}
-
-object Sink {
+  /**
+   * Ping the external resource to check it is healthy and ready to receive events
+   *
+   * The sink is expected to actively ping the external resource, e.g. by sending a HTTP request.
+   *
+   * This health check can yield any of three results:
+   *
+   *   1. `true` (wrapped in `F`). This is the only result that indicates the sink is healthy. 2.
+   *      `false` (wrapped in `F`). This means we successfully probed the external resource, but it
+   *      tells us the external resource is not ready to receive messages. 3. An exception raised
+   *      via the `F`.
+   *
+   * An application calling `pingForHealth` should check for all types of result: true / false /
+   * exception
+   */
+  def pingForHealth: F[Boolean]
```

Comment:
```suggestion
  def isHealthy: F[Boolean]
```
Sorry I know that this is annoying to receive comments about names. Generally I think that this is good practice to name functions that return a `Boolean` `isSomething`/`hasSomething`/ etc
It's free documentation for other developers